src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -389,17 +385,17 @@ private async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Block
             catch (ConsensusErrorException exception)
             {
                 this.logger.LogDebug("Peer's header is invalid. Peer will be banned and disconnected. Error: {0}.", exception.ConsensusError);
-                this.peerBanning.BanAndDisconnectPeer(peer.PeerEndPoint, this.connectionManager.ConnectionSettings.BanTimeSeconds, $"Peer presented invalid header, error: {exception.ConsensusError}.");
```

Comment:
If the only reason we  needed Connection Manager is for the settings then we could just inject the settings.